physical/cache.go

```diff
@@ -119,12 +119,13 @@ func (c *Cache) Put(ctx context.Context, entry *Entry) error {
 
 	lock := locksutil.LockForKey(c.locks, entry.Key)
 	lock.Lock()
-	defer lock.Unlock()
 
 	err := c.backend.Put(ctx, entry)
 	if err == nil {
 		c.lru.Add(entry.Key, entry)
 	}
+
+	lock.Unlock()
```

Comment:
Its possible that people submit PRs to put defer to use here. Or it might happen accidentally happen from folks within. Can we have a comment on each of these that they are indeed intentional?